sdk/iot/hub/src/az_iot_hub_client_methods.c

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include "az_iot_hub_client.h"
+#include <az_precondition_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+#include <_az_cfg.h>
+
+static const az_span methods_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("$iothub/methods/");
+static const az_span methods_topic_filter_suffix = AZ_SPAN_LITERAL_FROM_STR("POST/");
+static const az_span methods_response_topic_result = AZ_SPAN_LITERAL_FROM_STR("res/");
+static const az_span methods_response_topic_properties = AZ_SPAN_LITERAL_FROM_STR("/?$rid=");
+static const uint8_t hash_tag = '#';
+
+AZ_NODISCARD az_result az_iot_hub_client_methods_subscribe_topic_filter_get(
+    az_iot_hub_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter)
+{
+  AZ_PRECONDITION_NOT_NULL(client);
```

Comment:
missing : UNUSED(client). 
This will not compile if you disable preconditions.

Potentially in other places - please try to compile w/o preconditions by adding a 
```C
#define NO_PRECONDITION_CHECKING
```

at the top of this file.

Reply:
We have a gap in tests for this. Need to address that.
Thanks.